lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -1426,6 +1431,46 @@ newtype ApiMnemonicT (sizes :: [Nat]) =
     ApiMnemonicT { getApiMnemonicT :: SomeMnemonic }
     deriving (Generic, Show, Eq)
 
+-- | A stake key belonging to the current wallet.
+data ApiOurStakeKey n = ApiOurStakeKey
+     { _index :: !Natural
+    , _key :: !(ApiT W.RewardAccount, Proxy n)
+    , _stake :: !(Quantity "lovelace" Natural)
+      -- ^ The total ada this stake key controlls / is associated with. This
+      -- also includes the reward balance.
+    , _rewardBalance :: !(Quantity "lovelace" Natural)
+      -- ^ The current reward balance (not lifetime).
+    , _delegation :: !ApiWalletDelegation
+      -- ^ The delegation of this stake key
+    } deriving (Generic, Eq, Show)
+
+-- | A stake key found in the wallet UTxO, but which isn't ours.
+--
+-- We /could/ provide the current delegation status for foreign stake
+-- keys.
+data ApiForeignStakeKey n = ApiForeignStakeKey
+    { _key :: !(ApiT W.RewardAccount, Proxy n)
+    , _stake :: !(Quantity "lovelace" Natural)
+      -- ^ The total ada this stake key controlls / is associated with. This
+      -- also includes the reward balance.
+    , _rewardBalance :: !(Quantity "lovelace" Natural)
+      -- ^ The current reward balance (not lifetime).
+    } deriving (Generic, Eq, Show)
+
+-- | For describing how much stake is associated with no stake key.
+newtype ApiNullStakeKey = ApiNullStakeKey
+    { _stake :: Quantity "lovelace" Natural
+      -- ^ The total stake of the wallet UTxO that is not associated with a
+      -- stake key, because it's part of an enterprise address.
+    } deriving (Generic, Eq, Show)
+
+-- | Collection of stake keys associated with a wallet.
+data ApiStakeKeys n = ApiStakeKeys
+    { _ours :: ![ApiOurStakeKey n]
+    , _foreign :: ![ApiForeignStakeKey n]
+    , _null :: !ApiNullStakeKey
```

Comment:
`null` being a special keyword in JavaScript and JSON, I'd highly recommend to NOT using `null` as a field name! Perhaps `none` could work a bit better or `inactive` (though stake can be associated with a stake key and still be inactive.. so perhaps it is a poor choice).

Reply:
Ok, fair! Probably `none` then

Reply:
Another thing: I think currently the balance of say script and pointer addresses get included into this `none`, which could be really confusing if it happens...

`none` should probably only correspond to enterprise addresses. 

Then we could have another bucket for "other/unknown", or leave it out for now.

Reply:
> Then we could have another bucket for "other/unknown", or leave it out for now.

For now I just mentioned that `none` could include more things than enterprise addresses in the swagger API.